tests/test_config_reader.py

```diff
@@ -20,3 +22,13 @@ def test_parse_config_files():
         assert pytest_exit_object.type == SystemExit 
         assert pytest_exit_object.value.code == 42  
         assert pytest_exit_object.sys.stdout.getline().strip() == "No file named config.yml found in config directory {}. Exiting...".format(THIS_DIR)
+
+def test_get_output_params():
+    with mock.patch.dict(MOCK_CONFIG, {"general_parameters":{"data_output_filename":1.1}, "general_parameters":{"data_output_path":2.2}}):
```

Comment:
We don't have to mock the config file because this function simply takes a dictionary as input.  So you can simply give it a dictionary.  The code will be more readable. Also, this will allow us to use the paramaterize decorator (see my example) and we can give a bunch of different examples, some of which can test edge cases (for example, handing the function incorrect data).

Reply:
Ah yeah let me try implementing that. 

My first idea was to create a universal "good" config, and make alterations on that with the mock patches, but should be better with the paramaterize!

Reply:
Now passes dictionary through the paramaterize decorator.